release/pkg/assets/config/bundle_release.go

```diff
@@ -813,6 +815,32 @@ var bundleReleaseAssetsConfigMap = []assettypes.AssetConfig{
 	},
 }
 
-func GetBundleReleaseAssetsConfigMap() []assettypes.AssetConfig {
-	return bundleReleaseAssetsConfigMap
+func GetBundleReleaseAssetsConfigMap(rc *releasetypes.ReleaseConfig) []assettypes.AssetConfig {
+	conf := bundleReleaseAssetsConfigMap
+	if rc.DevRelease && rc.BuildRepoBranchName == "main" {
+		nutanixConfig := assettypes.AssetConfig{
+			ProjectName: "cluster-api-provider-nutanix",
+			ProjectPath: "projects/nutanix-cloud-native/cluster-api-provider-nutanix",
+			Images: []*assettypes.Image{
+				{
+					RepoName:  "manager",
+					AssetName: "cluster-api-provider-nutanix",
+				},
+			},
+			ImageRepoPrefix: "nutanix-cloud-native/cluster-api-provider-nutanix/release",
+			ImageTagOptions: []string{
+				"gitTag",
+				"projectPath",
+			},
+			Manifests: []*assettypes.ManifestComponent{
+				{
+					Name:          "infrastructure-nutanix",
+					ManifestFiles: []string{"infrastructure-components.yaml", "cluster-template.yaml", "metadata.yaml"},
+				},
+			},
+			OnlyForDevRelease: true,
+		}
+		conf = append(conf, nutanixConfig)
+	}
+	return conf
```

Comment:
we don't need a special check here since `OnlyForDevRelease` already handles this.
Can you move this section inside the `bundleReleaseAssetsConfigMap` array? 
It should be similar to this https://github.com/aws/eks-anywhere/blob/main/release/pkg/assets/config/bundle_release.go#L179-L200

Reply:
If you take a look at the test, the `release-0.11` is also used in the test as a "dev" release which is why I added the check for branch.

Reply:
If you take a look at TestGenerateBundleManifest,[ the release config](https://github.com/aws/eks-anywhere/blob/15a1cd2a1229abece283f24a2b499ee577b96006/release/pkg/operations/bundle_release_test.go#L59) has DevRelease set to true for both main and release-0.11 branch. As such, it tries to fetch a git tag for a project that doesn't exist in that branch and it fails.